src/main/java/dev/pgm/events/ready/ReadyListener.java

```diff
@@ -1,20 +1,35 @@
 package dev.pgm.events.ready;
 
+import dev.pgm.events.Tournament;
+import dev.pgm.events.config.AppData;
 import java.time.Duration;
+import java.util.Optional;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
 import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
 import org.bukkit.event.Listener;
+import tc.oc.pgm.api.match.MatchPhase;
 import tc.oc.pgm.api.match.event.MatchLoadEvent;
+import tc.oc.pgm.api.party.Party;
+import tc.oc.pgm.api.player.MatchPlayer;
 import tc.oc.pgm.events.CountdownCancelEvent;
 import tc.oc.pgm.events.CountdownStartEvent;
+import tc.oc.pgm.events.PlayerLeaveMatchEvent;
+import tc.oc.pgm.events.PlayerPartyChangeEvent;
+import tc.oc.pgm.match.ObserverParty;
 import tc.oc.pgm.start.StartCountdown;
 import tc.oc.pgm.start.StartMatchModule;
+import tc.oc.pgm.teams.Team;
 
 public class ReadyListener implements Listener {
 
+  private final ReadyManager manager;
   private final ReadySystem system;
   private final ReadyParties parties;
 
-  public ReadyListener(ReadySystem system, ReadyParties parties) {
+  public ReadyListener(ReadyManager manager, ReadySystem system, ReadyParties parties) {
```

Comment:
The `ReadyManager` already seems to contain a `ReadySystem` and a `ReadyParties`. I think it makes sense the manager can handle stuff related to the system and parties as well.

Reply:
Had this in my other branch but it turned in to a mess. I think the ready stuff needs a rework much bigger than this change.

Reply:
Ok